charmrepo/charmstore.go

```diff
@@ -71,7 +74,58 @@ func (s *CharmStore) Get(curl *charm.URL) (charm.Charm, error) {
 
 // Latest implements Interface.Latest.
 func (s *CharmStore) Latest(curls ...*charm.URL) ([]CharmRevision, error) {
-	return nil, notImplemented
+	if len(curls) == 0 {
+		return nil, nil
+	}
+
+	// Prepare the request to the charm store.
+	urls := make([]string, len(curls))
+	ids := make([]string, len(curls))
+	for i, curl := range curls {
+		url := curl.WithRevision(-1).String()
+		urls[i] = url
+		ids[i] = "id=" + url
+	}
+	path := "/meta/any?include=id-revision&include=hash256&" + strings.Join(ids, "&")
+
+	// Execute the request and retrieve results.
+	var results map[string]struct {
```

Comment:
I think you could avoid some work here by declaring
the retrieved fields explicitly:

    var results map[string] struct {
      Meta struct {
           IdRevision params.IdRevisionResponse `json:"id-revision"`
           Hash256 params.HashResponse `json:"hash256"`
      }
    }

Even better would be to add a bulk meta method as a companion
to csclient.Client.Meta.